airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java

```diff
@@ -30,15 +52,78 @@ public class SnowflakeDestinationHandler implements DestinationHandler<Snowflake
+                         """.formatted(paramHolder, paramHolder);
+    final String[] bindValues = new String[streamIds.size() * 2 + 1];
+    bindValues[0] = databaseName.toUpperCase();
+    System.arraycopy(namespaces, 0, bindValues, 1, namespaces.length);
+    System.arraycopy(names, 0, bindValues, namespaces.length + 1, names.length);
+    final List<JsonNode> results = database.queryJsons(query, bindValues);
+    for (final JsonNode result : results) {
+      final String tableSchema = result.get("TABLE_SCHEMA").asText();
+      final String tableName = result.get("TABLE_NAME").asText();
+      final String columnName = result.get("COLUMN_NAME").asText();
+      final String dataType = result.get("DATA_TYPE").asText();
+      final String isNullable = result.get("IS_NULLABLE").asText();
+      final TableDefinition tableDefinition = existingTables
+          .computeIfAbsent(tableSchema, k -> new LinkedHashMap<>())
+          .computeIfAbsent(tableName, k -> new TableDefinition(new LinkedHashMap<>()));
+      tableDefinition.columns().put(columnName, new ColumnDefinition(columnName, dataType, 0, fromIsNullableIsoString(isNullable)));
+    }
+    return existingTables;
+  }
+
+  private LinkedHashMap<String, LinkedHashMap<String, Integer>> getFinalTableRowCount(final List<StreamId> streamIds) throws SQLException {
+    final LinkedHashMap<String, LinkedHashMap<String, Integer>> tableRowCounts = new LinkedHashMap<>();
+    final String paramHolder = String.join(",", Collections.nCopies(streamIds.size(), "?"));
+    // convert list stream to array
+    final String[] namespaces = streamIds.stream().map(StreamId::finalNamespace).toArray(String[]::new);
+    final String[] names = streamIds.stream().map(StreamId::finalName).toArray(String[]::new);
+    final String query = """
+                         SELECT table_schema, table_name, row_count
+                         FROM information_schema.tables
+                         WHERE table_catalog = ?
+                           AND table_schema IN (%s)
+                           AND table_name IN (%s)
+                         """.formatted(paramHolder, paramHolder);
+    final String[] bindValues = new String[streamIds.size() * 2 + 1];
+    bindValues[0] = databaseName.toUpperCase();
+    System.arraycopy(namespaces, 0, bindValues, 1, namespaces.length);
+    System.arraycopy(names, 0, bindValues, namespaces.length + 1, names.length);
+    final List<JsonNode> results = database.queryJsons(query, bindValues);
+    for (final JsonNode result : results) {
+      final String tableSchema = result.get("TABLE_SCHEMA").asText();
+      final String tableName = result.get("TABLE_NAME").asText();
+      final int rowCount = result.get("ROW_COUNT").asInt();
+      tableRowCounts.computeIfAbsent(tableSchema, k -> new LinkedHashMap<>()).put(tableName, rowCount);
+    }
+    return tableRowCounts;
+  }
+
+  public Optional<TableDefinition> findExistingTable(final StreamId id) throws SQLException {
```

Comment:
can we rewrite this as just `return findExistingTables(List.of(id)).get(id)`?

Reply:
Unified it in SnowflakeV2Migrator.